### PR TITLE
Fix: truncate table param error && faster refreshing of the routing table for specific error code

### PR DESCRIPTION
**Pull request by maochongxin**
1. New Exception Type for RoutingWrong Error:
  This allows for faster refreshing of the routing table when this specific error occurs.
2. Fix `tabletId`(partitionId) Issue After Route Refresh:
  Resolved an issue where the `tabletId` was not being set after refreshing the routing table.
This fix ensures that retries do not fail due to wrong `tabletId`.

### CODE REVIEW

File: src/main/java/com/alipay/oceanbase/rpc/bolt/transport/ObTableRemoting.java [hunk 1]
```diff
@@ -132,7 +121,17 @@ public ObPayload invokeSync(final ObTableConnection conn, final ObPayload reques
...
             ObRpcResultCode resultCode = new ObRpcResultCode();
             resultCode.decode(buf);
-
+            // If response indicates the request is routed to wrong server, we should refresh the routing meta.
+            if (response.getHeader().isRoutingWrong()) {
+                String errMessage = TraceUtil.formatTraceMessage(conn, request,
+                        "routed to the wrong server: " + response.getMessage());
+                logger.warn(errMessage);
+                if (needFetchAll(resultCode.getRcode())) {
+                    throw new ObTableRoutingWrongException(errMessage);
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/bolt/transport/ObTableRemoting.java:130]: use another new exception?   this exception has been used before and cannot represent that needs to refresh all table entry

File: src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java [hunk 2]
```diff
@@ -380,6 +382,12 @@ public void partitionExecute(ObTableOperationResult[] results,
                                 "tablename:{} partition id:{} batch ops retry while meet ObTableMasterChangeException, errorCode: {} , retry times {}",
                                 tableName, partId, ((ObTableException) ex).getErrorCode(),
                                 tryTimes, ex);
+                        if (ex instanceof ObTableRoutingWrongException) {
+                            System.out.println("need fetchAll refresh");
+                            obTableClient.getOrRefreshTableEntry(tableName, true, obTableClient.isTableEntryRefreshIntervalWait(), true);
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java:387]: getTable will call getOrRefreshTableEntry once, which means getOrRefreshTableEntry will be called twice? Can we merge into one getOrRefreshTableEntry?

File: src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java [hunk 3]
```diff
@@ -380,6 +382,12 @@ public void partitionExecute(ObTableOperationResult[] results,
                                 "tablename:{} partition id:{} batch ops retry while meet ObTableMasterChangeException, errorCode: {} , retry times {}",
                                 tableName, partId, ((ObTableException) ex).getErrorCode(),
                                 tryTimes, ex);
+                        if (ex instanceof ObTableRoutingWrongException) {
+                            System.out.println("need fetchAll refresh");
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java:386]: should use logger instead of print directly

**Review by shenyunlong — CHANGES_REQUESTED** [review]

File: src/main/java/com/alipay/oceanbase/rpc/bolt/transport/ObTableRemoting.java (hunk 1 above)
Comment (maochongxin) [reply src/main/java/com/alipay/oceanbase/rpc/bolt/transport/ObTableRemoting.java:135]: Fixed

File: src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java (hunk 2 above)
Comment (maochongxin) [reply src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java]: Fixed

**Comment by shenyunlong** [comment]
LGTM

**Pull request merged**